src/accounts.rs

```diff
@@ -318,17 +326,50 @@ struct InnerConfig {
     pub accounts: Vec<AccountConfig>,
 }
 
+impl Drop for Config {
+    fn drop(&mut self) {
+        if let Some(lock_task) = self.lock_task.take() {
+            lock_task.abort();
+            futures::executor::block_on(lock_task).ok();
+        }
+    }
+}
+
 impl Config {
-    /// Creates a new configuration file in the given account manager directory.
-    pub async fn new(dir: &Path) -> Result<Self> {
+    /// Creates a new Config for `file`, but doesn't open/sync it.
+    async fn new_nosync(file: PathBuf) -> Result<Self> {
+        let dir = file.parent().context("can't get config file directory")?;
         let inner = InnerConfig {
             accounts: Vec::new(),
             selected_account: 0,
             next_id: 1,
         };
-        let file = dir.join(CONFIG_NAME);
-        let mut cfg = Self { file, inner };
+        let lockfile = dir.join(LOCKFILE_NAME);
+        let mut lock = fd_lock::RwLock::new(fs::File::create(lockfile).await?);
+        let (locked_tx, locked_rx) = oneshot::channel();
+        let lock_task: JoinHandle<anyhow::Result<()>> = tokio::spawn(async move {
+            let _guard = lock.try_write()?;
```

Comment:
`fd_lock` crate only has ugly RAII guards, moreover non-owned (unlike tokio f.e.), so i do the job in a separate task, otherwise `Config` becomes a self-referential struct.

I think that having only RAII guards is an antipattern, it only brings problems when a user needs to implement their own guards on top of the locking logic.